releasenotes/notes/const-expr-397ff09042942b81.yaml

```diff
@@ -0,0 +1,24 @@
+---
+features_circuits:
+  - |
+    The classical realtime-expressions module :mod:`qiskit.circuit.classical` can now represent
+    constant expressions. The :class:`~.expr.Expr` class now has a bool
+    :attr:`~.expr.Expr.const` attribute which indicates the expression's const-ness. This allows
+    us to enforce that expressions in certain contexts must be possible to evaluate at compile-time.
+
+    All :class:`~.expr.Var` expressions are considered to be non-const, while all :class:`~.expr.Value`
+    expressions are const.
+
+    An expression comprised only of other const expressions is also const::
+
+      from qiskit.circuit.classical import expr
+
+      expr.bit_and(5, 6).const
+      # True
```

Comment:
```suggestion
      assert expr.bit_and(5, 6).const
```

Reply:
This is stylistic choice, I just suggested it like this because I think the `expr` documentation does things like this, and it would (in theory) then immediately play nicely with automated running of the code snippets.

Feel free to ignore if you prefer anyway.